handlers/Subscribe.cpp

```diff
@@ -0,0 +1,233 @@
+        }
+    }
+
+    return nullptr;
+}
+
+void
+subscribeToAccounts(
+    boost::json::object const& request,
+    std::shared_ptr<session>& session,
+    SubscriptionManager& manager)
+{
+    boost::json::array const& accounts = request.at("accounts").as_array();
+
+    for (auto const& account : accounts)
+    {
+        std::string s = account.as_string().c_str();
+
+        auto accountID = ripple::parseBase58<ripple::AccountID>(s);
+
+        if(!accountID)
+        {
+            assert(false);
+            continue;
+        }
+
+        manager.subAccount(*accountID, session);
+    }
+}
+
+void
+unsubscribeToAccounts(
+    boost::json::object const& request,
+    std::shared_ptr<session>& session,
+    SubscriptionManager& manager)
+{
+    boost::json::array const& accounts = request.at("accounts").as_array();
+
+    for (auto const& account : accounts)
+    {
+        std::string s = account.as_string().c_str();
+
+        auto accountID = ripple::parseBase58<ripple::AccountID>(s);
+
+        if(!accountID)
+        {
+            assert(false);
+            continue;
+        }
+
+        manager.unsubAccount(*accountID, session);
+    }
+}
+
+boost::json::object
+doSubscribe(
+    boost::json::object const& request,
+    std::shared_ptr<session>& session,
+    SubscriptionManager& manager)
+{
+    boost::json::object response;
+
+    if (request.contains("streams"))
+    {
+        boost::json::value error = validateStreams(request);
+
+        if(!error.is_null())
+        {
+            response["error"] = error;
+            return response;
+        }   
+    }
+
+    if (request.contains("accounts"))
+    {
+        boost::json::value error = validateAccounts(request);
+
+        if(!error.is_null())
+        {
+            response["error"] = error;
+            return response;
+        }  
+    }
+
+    if (request.contains("streams"))
+        subscribeToStreams(request, session, manager);
+
+    if (request.contains("accounts"))
+        subscribeToAccounts(request, session, manager);
+
+    response["status"] = "success";
+    return response;
+}
+
+boost::json::object
+doUnsubscribe(
+    boost::json::object const& request,
+    std::shared_ptr<session>& session,
+    SubscriptionManager& manager)
+{
+    boost::json::object response;
+
+    if (request.contains("streams"))
+    {
+        boost::json::value error = validateStreams(request);
+
+        if(!error.is_null())
+        {
+            response["error"] = error;
+            return response;
+        }   
+    }
+
+    if (request.contains("accounts"))
+    {
+        boost::json::value error = validateAccounts(request);
+
+        if(!error.is_null())
+        {
+            response["error"] = error;
+            return response;
+        }  
+    }
+
+    if (request.contains("streams"))
+        unsubscribeToStreams(request, session, manager);
+
+    if (request.contains("accounts"))
+        unsubscribeToAccounts(request, session, manager);
```

Comment:
Let's combine the `if`s here, so there are two instead of four.

Reply:
So here, if we combine the `if`s,  then there's a chance you get a `subscribe` request w/ an error in accounts, that successfully subscribes to streams, but then returns a failure code because accounts failed. here we are validating both streams and accounts, and then subscribing.